app/src/main/java/com/walkbase/ibeacon/sdk/IBeacon.kt

```diff
@@ -0,0 +1,216 @@
+package com.walkbase.ibeacon.sdk
+
+import android.Manifest
+import android.bluetooth.le.AdvertiseSettings
+import android.content.pm.PackageManager
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.IntDef
+import androidx.core.content.ContextCompat
+import com.walkbase.ibeacon.R
+import org.altbeacon.beacon.Beacon
+import org.altbeacon.beacon.BeaconParser
+import org.altbeacon.beacon.BeaconTransmitter
+
+// https://www.bluetooth.com/specifications/assigned-numbers/
+private const val APPLE_INC_MANUFACTURER_ID: Int = 0x004C
+
+// https://github.com/AltBeacon/android-beacon-library/issues/13
+private const val IBEACON_LAYOUT = "m:2-3=0215,i:4-19,i:20-21,i:22-23,p:24-24,d:25-25"
+private const val IBEACON_TYPE_CODE = 0x0215
```

Comment:
This took the longest to understand and debug. It took a while to find the answer [here](https://altbeacon.github.io/android-beacon-library/beacon-transmitter.html) and [here](https://github.com/AltBeacon/android-beacon-library/issues/13).